packages/ai/src/providers/google/index.ts

```diff
@@ -66,26 +66,40 @@ const createChatCompletion: ICreateChatCompletion = async (body) => {
   const { model, messages } = body;
   const googleAI = new GoogleGenerativeAI(process.env.GOOGLE_API_KEY!);
   const timeout = body.timeout || DEFAULT_TIMEOUT;
-  const modelInstance = googleAI.getGenerativeModel(
-    { model },
-    { apiVersion: "v1beta", timeout },
-  );
+  const isBetaModel = model.includes("gemini-1.5");
+  // Google's JS library does not fully support Gemini 1.5 Pro
+  // We have an open issue with details:
+  // https://github.com/google/generative-ai-js/issues/98
+  const modelInstance = isBetaModel
+    ? googleAI.getGenerativeModel({
+        model: "gemini-pro",
+      })
+    : googleAI.getGenerativeModel({ model }, { apiVersion: "v1beta", timeout });
   const contents = massageOpenAIMessagesToGoogleAI(messages);
   const { executionDone } = await batch.waitForTurn();
   try {
     const startedAt = Date.now();
     const completion = await promiseRetry<GenerateContentResult>(
       (retry) => {
-        // TODO: move to model.startChat which support model config (e.g. temperature)
-        return modelInstance
-          .generateContent({ contents })
-          .catch((err: Error) => {
-            // TODO: Replace with instanceof checks when the Gemini SDK exports errors
-            if (err.message.includes("[429 Too Many Requests]")) {
-              retry(err);
-            }
-            throw err;
-          });
+        return (
+          isBetaModel
+            ? modelInstance.generateContent({ contents })
```

Comment:
would json mode on gemini 1.5 pro work

Reply:
see example here: https://github.com/google-gemini/cookbook/blob/main/quickstarts/rest/JSON_mode_REST.ipynb

Reply:
<img width="1138" alt="image" src="https://github.com/empirical-run/empirical/assets/57426646/26754ee4-9380-45b9-9d8e-9ac17b8d1365">
Not supported

Reply:
Valid properties:

```
export declare interface GenerationConfig {
    candidateCount?: number;
    stopSequences?: string[];
    maxOutputTokens?: number;
    temperature?: number;
    topP?: number;
    topK?: number;
}
```

Reply:
haan looks like the JS library is broken. the REST endpoint is working with it - so it's actually supported under the hood